{{cookiecutter.package_name}}/{{cookiecutter.import_name}}/app/jupyter.py

```diff
@@ -0,0 +1,35 @@
+import asyncio
+
+from trame import port
+from trame.jupyter import display_iframe
+
+from {{cookiecutter.import_name}}.app import controller, ui
+
+
+def run_in_loop(**kwargs):
+    """Run and display the trame application in jupyter's event loop
+
+    The kwargs are forwarded to IPython.display.IFrame()
+    """
+    controller.on_start()
+    ui.layout.start(exec_mode="task", server=True)
```

Comment:
FYI: one problem with having these two lines of code here is that we only want to run these two lines of code once. So if you re-run the `run_in_loop()` function again, we get some exceptions.

Reply:
you can probably guard them with a global flag within that module

Reply:
Done.